geode-core/src/main/java/org/apache/geode/internal/cache/PoolFactoryImpl.java

```diff
@@ -339,15 +340,36 @@ public Pool create(String name) throws CacheException {
     InternalDistributedSystem distributedSystem = InternalDistributedSystem.getAnyInstance();
     InternalCache cache = GemFireCacheImpl.getInstance();
     ThreadsMonitoring threadMonitoring = null;
+
     if (cache != null) {
       threadMonitoring = cache.getDistributionManager().getThreadMonitoring();
       TypeRegistry registry = cache.getPdxRegistry();
       if (registry != null && !attributes.isGateway()) {
         registry.creatingPool();
       }
     }
-    return PoolImpl.create(this.pm, name, this.attributes, this.locatorAddresses, distributedSystem,
-        cache, threadMonitoring);
+
+    Pool returnPool = null;
+    int retries = 0;
+    final int MAX_RETRIES = 20;
+    while (returnPool == null && retries < MAX_RETRIES) {
+      try {
+        returnPool =
+            PoolImpl
+                .create(this.pm, name, this.attributes, this.locatorAddresses, distributedSystem,
+                    cache, threadMonitoring);
+        logger.info(" **** Did we retry?");
```

Comment:
Please reword this. The overall guideline is that any log statement above debug should make sense to a customer.

Reply:
This is still in debug... Not ready for merge...